src/jackdaw/serdes/avro.clj

```diff
@@ -389,15 +389,17 @@
 (defrecord RecordType [^Schema schema field->schema+coercion]
   SchemaCoercion
   (match-clj? [_ clj-map]
-    (let [[_ unknown-fields _] (clojure.data/diff (set (keys field->schema+coercion))
-                                                  (set (keys clj-map)))]
-      (and (every? (fn [[field-key [^Schema$Field field field-coercion]]]
-                     (let [field-value (get clj-map field-key ::missing)]
-                       (if (= field-value ::missing)
-                         (.defaultValue field)
-                         (match-clj? field-coercion field-value))))
-                   field->schema+coercion)
-           (empty? unknown-fields))))
+    (if (map? clj-map)
```

Comment:
Since the `else` clause returns false, I think this could maybe just be `(when (map? clj-map) ...)`.

Reply:
@cddr That's what I had initially, but then I noticed explicit `(boolean ..` forms in other `match-clj?` implementations, and so thought maybe returning falsey `nil` was against the contract. If not glad to change!

Reply:
Made this change and also updated `CHANGELOG`. Tried to re-request review but the button isn't doing anything. 🤷🏼‍♂️